### ISSUE TITLE
write_dct and write_spatial leak file handles

### ISSUE THREAD

**Issue by timcowlishaw** [martinbenes1996/jpeglib#8]
Hi there Martin, thanks a lot for this library - I have a project in which i have to manipulate JPEG DCTs directly and thought i was going to have to do something  really low-level myself 'til i found it :-)

I've noticed a slight problem however, on my machine (Mac OSX 11.6.8, python 3.68, jpeglib 0.12.5, libjpeg 9e) it looks like the `write_dct` and `write_spatial` methods are leaking file descriptors, the following method (for example):
```
def decay_jpeg_dct_y_absolute(filename):
    try:
        print("decaying dct luminance")
        im = jpeglib.read_dct(filename)
        original = im.Y
        min = np.min(im.Y)
        max = np.max(im.Y)
        perturbed = perturb_array(original, min=min, max=max)
        im.Y = perturbed
        im.write_dct()
        im.close()
    except OSError:
        pass
```
...will leave a reference to `filename` open (verified with lsof) - this only happens when `write_dct` is called, so it doesn't appear to be related to the file reading. `write_spatial` also exhibits the same behaviour. 

I've had a look into the code with a view to opening a PR, but the actual file writing seems to be in C++ land which is far beyond my abilities i'm afraid!

Any chance of a fix, or a pointer about what i might be doing wrong?

Many thanks!

Tim

**Comment by timcowlishaw** [comment]
Sorry Martin, false alarm, after looking back at the code block in the issue - i suddently realised that i should look back at my rather hacky exception handling, and that turned out to be the problem! Nothing wrong with your library at all. Apologies!

Tim

**Comment by timcowlishaw** [comment]
Aha, after a bit of further investigation, perhaps the problem is in two places - when the image cannot be opened and an exception is raised from within the library (`OSException` with message:  `reading info of $FILENAME failed`) the file descriptor isn't closed, but as there's no reference to the file available outside the library, it can't be closed in exception handling

**Comment by martinbenes1996** [comment]
Well spotted ;) This was unintentionally introduced by the solution to #4.
Have to think a bit, about how to solve this.

**Comment by martinbenes1996** [comment]
The suggested solution is now being developed in the dev branch.

https://github.com/martinbenes1996/jpeglib/commit/7082853a319d2b48e79a8850b4663a0085b80e6f

**Comment by martinbenes1996** [comment]
The fix is part of release 0.12.7.
https://github.com/martinbenes1996/jpeglib/releases/tag/0.12.7

**Comment by timcowlishaw** [comment]
aaah amazing, thank you so much Martin!
